src/app.ts

```diff
@@ -1,19 +1,23 @@
-﻿import debug = require('debug');
-import * as express from "express";
+﻿import * as express from "express";
+import "reflect-metadata";
 import { Configuration } from "./Common/Configuration";
-//import path = require('path');
+import { IDataProvider } from "./DataProviders/IDataProvider";
+import { myContainer } from "./inversify.config";
+import { HomeController } from './Routes/HomeController';
+import { SeasonsController } from "./Routes/SeasonsController";
+import { TYPES } from "./types";
+import debug = require('debug');
 
 Configuration.initialize();
-
-import routes from './routes/index';
-import seasons from "./routes/seasons";
-
+const dataProvider: IDataProvider = myContainer.get<IDataProvider>(TYPES.IDataProvider);
+const homeController = new HomeController();
+const seasonsController = new SeasonsController(dataProvider);
```

Comment:
I would suggest making these injectable as well, and getting their instances through `myContainer.get(..)`. This will mean that the controllers are free to change their dependencies without your `app.ts` file having to change. Currently they're unnecessarily tightly coupled.

Reply:
I don't think I want the controllers to be injectables as well...I don't see a need.  The plan for testing is to inject a data provider.

Reply:
The goal of making them injectable is not necessarily just to test them, but to make them free to change their dependencies without having to modify app.ts. Essentially it means only one thing needs to know how to create them (the container) rather than two.